src/trio/_core/_run.py

```diff
@@ -1706,6 +1706,7 @@ def close(self) -> None:
         self.asyncgens.close()
         if "after_run" in self.instruments:
             self.instruments.call("after_run")
+        self.system_nursery: Nursery | None = None
```

Comment:
Type hint here isn't necessary

Reply:
As long as this isn't the first time `system_nursery` is set yes